### PR TITLE
Graph loading bug

### PR DESCRIPTION
**Pull request by JGSweets**
Allows lazy loading to work for data class as other classes without causing bugs

### CODE REVIEW

File: dataprofiler/data_readers/graph_data.py [hunk 1]
```diff
@@ -102,25 +104,19 @@ def _find_target_string_in_column(self, column_names, keyword_list):
         return target_index
 
     @classmethod
-    def csv_column_names(cls, file_path, options):
+    def csv_column_names(cls, file_path, header, delimiter, encoding="utf-8"):
```
Comment (JGSweets) [inline dataprofiler/data_readers/graph_data.py:107]: updated for reusability in _network

File: dataprofiler/data_readers/graph_data.py [hunk 2]
```diff
@@ -173,15 +174,58 @@ def _format_data_networkx(self):
...
-        csv_as_list = []
-        if not self._checked_header or self._header == "auto":
+        if not self._checked_header or not self._delimiter:
+            delimiter, quotechar = None, None
             data_as_str = data_utils.load_as_str_from_file(
                 self.input_file_path, self.file_encoding
             )
-            self._header = CSVData._guess_header_row(
-                data_as_str, self._delimiter, self._quotechar
+
```
Comment (JGSweets) [inline dataprofiler/data_readers/graph_data.py:182]: this section is what fixes the bug. If graph data was lazily loaded or loaded without options from Is_match, it wouldn't determine these when thrown into the system. This fixes it.

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 3]
```diff
@@ -98,6 +98,11 @@ def setUpClass(cls):
 
         cls.file_or_buf_list = cls.input_file_names_pos + cls.buffer_list
 
+    @classmethod
+    def setUp(cls):
+        for buffer in cls.buffer_list:
+            buffer["path"].seek(0)
+
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_graph_data.py:105]: need to seek to reset data files after use between tests

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 4]
```diff
@@ -147,11 +152,17 @@ def test_csv_column_names(self):
             "open_date_src",
             "open_date_dst",
         ]
-        for input_file in self.input_file_names_pos:
+        for input_file in self.file_or_buf_list:
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_graph_data.py:155]: should be using all data for tests

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 5]
```diff
@@ -147,11 +152,17 @@ def test_csv_column_names(self):
...
             if input_file["list_nodes"] is None or input_file["list_edges"] is None:
                 continue
+            options = input_file["options"]
             self.assertEqual(
-                GraphData.csv_column_names(input_file["path"], input_file["options"]),
+                GraphData.csv_column_names(
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_graph_data.py:160]: fixes to new call

File: dataprofiler/tests/data_readers/test_csv_graph_data.py [hunk 6]
```diff
@@ -207,17 +218,17 @@ def test_data_loader_edges(self):
             self.assertTrue(all_edges_present)
 
     def test_class_load(self):
-        for input_file in self.input_file_names_pos:
+        for input_file in self.file_or_buf_list:
             data = GraphData(input_file["path"])
-            self.assertEqual(type(data), GraphData)
+            self.assertIsNotNone(data.data)
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_graph_data.py:223]: ensures data can be loaded as expected

Comment (taylorfturner) [reply dataprofiler/tests/data_readers/test_csv_graph_data.py:223]: I don’t mind both but the new is better at validating

**Review by taylorfturner — DISMISSED** [review]

**Comment by taylorfturner** [comment]
Pre commit too

**Review by taylorfturner — APPROVED** [review]

**Review by taylorfturner — CHANGES_REQUESTED** [review]
L250-L252 is actually failing here when I run the notebook @JGSweets

**Comment by taylorfturner** [comment]
> data_readers/graph_data.py L250-L252 is actually failing here when I run the notebook @JGSweets

False alarm -- we good

**Review by ksneab7 — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]

**Pull request merged**